.github/workflows/publish-package.yaml

```diff
@@ -0,0 +1,61 @@
+name: Publish Python Package
+on:
+  push:
+    branches:
+      - main
+    paths: # assume to only publish package when pyproject.toml file changes
+      - pyproject.toml
+  
+env: # default versions
+  POETRY_VERSION: 1.3.1
+  PYTHON_VERSION: 3.11.2
```

Comment:
Ideally both these env vars, and the values used in the "Install Poetry" step above, can come from files in the repository: we typically maintain `.python-version` and `.poetry-version` files for this reason. Is it possible to source these config options from the same project file in all cases?

Reply:
I've already added a step that attempts to read the .python-version file (if it exists). If there is a .python-version file, the default PYTHON_VERSION variable will be overwritten.

I can add the same logic for setting the POETRY_VERSION.

I'm hesitant to soley rely on .python-version and .poetry-version files b/c I know some repos don't have .poetry-version files.